src/runtime/linux_host_cpu_count.cpp

```diff
@@ -3,8 +3,27 @@
 extern "C" {
 
 extern long sysconf(int);
+extern int sched_getaffinity(int32_t pid, size_t cpusetsize, unsigned long *cpuset);
 
 WEAK int halide_host_cpu_count() {
-    return sysconf(84);
-}
+    unsigned long count = 0, cpuset[8];  // enough bits for 256 CPUs on 32 bit, or 512 CPUs on 64 bit
+    memset(cpuset, 0, sizeof(cpuset));
+    int rv = sched_getaffinity(0, sizeof(cpuset), cpuset);
```

Comment:
Doesn't this only return the affinity of the calling thread? If someone were to write a program that set the affinity of thread 0 to 0, thread 1 to 1, ..., thread N to N (something I've thought about doing but never done), and then called halide_host_cpu_count on one of those threads, I think it would return 1, which seems wrong.

Reply:
Possibly.  But I don't think that's a valid use case.  Please correct me if I'm wrong.

I think the Halide runtime uses this function to determine how many worker threads to spawn.  Any other threads you spawn outside of Halide may have their own affinities, but Halide won't ever see or use those.  Halide will only run on the thread that calls it, and any threads spawned by that one, which should inherit the affinity of that one.  Correct?

So let's say you fuss with thread affinities before calling the Halide runtime, setting the affinity of that thread to 1 single CPU.  Then, any additional threads spawned by the Halide runtime will be stuck on that one CPU.  Do you really want to be spawning 64 threads in that case?  I think 1 is a more sensible answer in that case.  Halide was given one CPU to work with, so it should run with one thread.